src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/Snapshots/VsProjectRestoreInfo.cs

```diff
@@ -23,9 +23,9 @@ public VsProjectRestoreInfo(ProjectRestoreInfo info)
 
     public string BaseIntermediatePath => _info.MSBuildProjectExtensionsPath;
 
-    public IVsTargetFrameworks2 TargetFrameworks => _targetFrameworks ??= new VsTargetFrameworks(_info.TargetFrameworks);
+    public IReadOnlyList<IVsTargetFrameworkInfo4> TargetFrameworks => _targetFrameworks ??= ImmutableList.CreateRange(_info.TargetFrameworks.Select(r => new VsTargetFrameworkInfo(r)));
```

Comment:
`ImmutableList` is a linked list, which uses more memory than an array-backed list, and has worse locality of reference. Please just use `List<>` here, or if you are really concerned about modification then `ImmutableArray<>` (however that will box when cast to `IROL<>`).

Ditto below.